Orange/base.py

```diff
@@ -65,13 +106,21 @@ def __call__(self, data):
         return model
 
     def preprocess(self, data):
-        """
-        Apply the `preprocessors` to the data.
-        """
-        for pp in self.preprocessors:
+        """Apply the `preprocessors` to the data"""
+        for pp in self.active_preprocessors:
             data = pp(data)
         return data
 
+    @property
+    def active_preprocessors(self):
+        # Check if preprocessors were overriden
+        if self.preprocessors is not type(self).preprocessors:
+            yield from self.preprocessors
+            if self.use_default_preprocessors:
+                yield from tuple(type(self).preprocessors)
+        else:
+            yield from type(self).preprocessors or ()
```

Comment:
Really please excuse me, but I find this slightly unintuitive to follow. Isn't it functionally the same, but more readable:
```py
# yield instance preprocessors first. they exist and are definitely a tuple, as ensured by the constructor
yield from self.preprocessors
# if the instance preprocessors aren't the same as class', apply the latter also
if (self.use_default_preprocessors
        and self.preprocessors is not type(self).preprocessors)):
    yield from type(self).preprocessors
```
Yielding from an empty tuple is, of course, a non-issue.